core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/JavascriptSettingProducer.java

```diff
@@ -0,0 +1,141 @@
+
+import org.silverpeas.core.util.SilverpeasBundle;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.silverpeas.core.util.EncodeHelper.javaStringToJsString;
+
+/**
+ * This tool permits the creates dynamically javascript settings that javascript plugins can used
+ * easily.<br/>
+ * The generated javascript settings will be an instance of SilverpeasPluginSettings defined into
+ * silverpeas.js
+ * @author Yohann Chastagnier
+ */
+public class JavascriptSettingProducer {
+
+  private String jsSettingVariableName = "Settings";
+  private Map<String, String> keySettings = new LinkedHashMap<>();
+
+  /**
+   * Initializes the setting producer by specifying the name of the javascript variable that
+   * represents the settings.
+   * @param jsSettingVariableName the javascript variable name of the bundle.
+   * @return the initialized producer instance.
+   */
+  public static JavascriptSettingProducer settingVariableName(final String jsSettingVariableName) {
+    return new JavascriptSettingProducer().withSettingVariableName(jsSettingVariableName);
+  }
+
+  /**
+   * Initializes the setting producer by specifying the name of the javascript variable that
+   * represents the settings.
+   * @param jsSettingVariableName the javascript variable name of the bundle.
+   * @return itself.
+   */
+  private JavascriptSettingProducer withSettingVariableName(final String jsSettingVariableName) {
+    this.jsSettingVariableName = jsSettingVariableName;
+    return this;
+  }
+
+  /**
+   * Adds given keys from the given bundle.
+   * @param bundle the bundle from which the messages must be extracted.
+   * @param keys the requested message keys.
+   * @return itself.
+   */
+  public JavascriptSettingProducer add(final SilverpeasBundle bundle, final String... keys) {
+    for (String key : keys) {
+      add(key, bundle.getString(key));
+    }
+    return this;
+  }
+
+  /**
+   * Adds given key / value.
+   * @param key the key to add.
+   * @param value the boolean value associated to the key.
+   * @return itself.
+   */
+  public JavascriptSettingProducer add(final String key, final String value) {
+    keySettings.put(key, "\"" + javaStringToJsString(value) + "\"");
+    return this;
+  }
+
+  /**
+   * Adds given key / value.
+   * @param key the key to add.
+   * @param value the boolean value associated to the key.
+   * @return itself.
+   */
+  public JavascriptSettingProducer add(final String key, final boolean value) {
+    keySettings.put(key, String.valueOf(value));
+    return this;
+  }
+
+  /**
+   * Adds given key / value.
+   * @param key the key to add.
+   * @param value the number value associated to the key.
+   * @return itself.
+   */
+  public JavascriptSettingProducer add(final String key, final Number value) {
+    keySettings.put(key, String.valueOf(value));
+    return this;
+  }
+
+  /**
+   * Hidden constructor.
+   */
+  private JavascriptSettingProducer() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1213)